### PR TITLE
[PORT] port invoice_validation_wkfl to v8

### PR DESCRIPTION
**Pull request by rdeheele**
rename views to remove partner name
partially port with new API
fix the transition between draft and cancel states

### CODE REVIEW

**Comment by yvaucher** [comment]
Some PEP8 and pylint issues to fix

**Comment by yvaucher** [comment]
@rdeheele It seems you forgot to git mv the original files

**Comment by rdeheele** [comment]
thanks @yvaucher , I'll work on it again later today

File: invoice_validation_wkfl/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Joël Grand-Guillaume (Camptocamp)
+#    Copyright 2010 Camptocamp SA
```
Comment (yvaucher) [inline invoice_validation_wkfl/__openerp__.py]: You can also bump the copyright year to 2010-2015

File: invoice_validation_wkfl/invoice.py [hunk 2]
```diff
@@ -18,20 +18,22 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 ##############################################################################
-from openerp.osv import orm, fields
-from tools.translate import _
-import netsvc
+from openerp import models, fields, api
+from openerp.osv import orm
```
Comment (yvaucher) [inline invoice_validation_wkfl/invoice.py]: travis will certainly warn about it, but this import is useless

File: invoice_validation_wkfl/invoice.py [hunk 3]
```diff
@@ -39,27 +41,27 @@ def action_to_valid(self, cr, uid, ids):
                     str_error_lines += "\n- %s" % line.name
                     errors = True
             if errors:
-                raise orm.except_orm(
+                raise except_osv(
```
Comment (yvaucher) [inline invoice_validation_wkfl/invoice.py]: `osv.except_osv` can be replaced by `openerp.exceptions.Warning`

**Comment by yvaucher** [comment]
:+1: Thanks

**Comment by pedrobaeza** [comment]
You can take the occassion to rename the module to a more generic name: _account_invoice_validation_workflow_.

**Comment by rdeheele** [comment]
@pedrobaeza good idea!

**Comment by pedrobaeza** [comment]
Please generate also POT file.

**Comment by lepistone** [comment]
:+1: , and you addressed @pedrobaeza's remarks. Thanks!

**Comment by pedrobaeza** [comment]
Please check https://github.com/OpenUpgrade/OpenUpgrade/pull/221 for a proper migration path for this module.

**Pull request merged**